File: client/java/src/avt/ParkService.java
package avt;

import java.io.IOException;
import main.Canvas;
import main.GameMidlet;

public final class ParkService extends IService {
   private static ParkService instance;

   public static ParkService gI() {
      if (instance == null) {
         instance = new ParkService();
      }

      return instance;
   }

   public final void doJoinPark(int var1, int var2) {
      System.out.println("doJoinPark: " + var1 + "   " + var2);
      if (OnScreen.c == 0) {
         Canvas.startWaitDlg();
      }

      this.createMessage((byte)50);

      try {
         super.m.writer().writeByte(var1);
         super.m.writer().writeByte(var2);
         super.m.writer().writeShort(LoadMap.xDichChuyen_);
         super.m.writer().writeShort(LoadMap.C);
      } catch (IOException var4) {
         var4.printStackTrace();
      }

      this.sendMessage();
   }

   public final void doMove(int var1, int var2, int var3, int var4) {
      this.createMessage((byte)54);

      try {
         super.m.writer().writeShort(var1);
         super.m.writer().writeShort(var2);
         super.m.writer().writeByte(var3);
         super.m.writer().writeShort(var4);
      } catch (IOException var6) {
      }

      this.sendMessage();
   }

   public final void chatToBoard(String var1) {
      if (GameMidlet.CLIENT_TYPE == 10) {
         this.createMessage((byte)77);
      } else {
         this.createMessage((byte)55);
      }

      this.writeUTF(var1);
      this.sendMessage();
   }

   public final void doRequestAddFriend(int var1) {
      this.createMessage((byte)-21);
      this.writeInt(var1);
      this.sendMessage();
   }

   public final void doAddFriend(int var1, boolean var2) {
      this.createMessage((byte)-19);

      try {
         super.m.writer().writeInt(var1);
         super.m.writer().writeBoolean(var2);
      } catch (IOException var4) {
      }

      this.sendMessage();
   }

   public final void doGiftGiving(int var1, int var2, int var3) {
      System.out.println("doGiftGiving: " + var1 + "    " + GameMidlet.avatar.IDDB);
      this.createMessage((byte)58);

      try {
         super.m.writer().writeInt(var1);
         super.m.writer().writeShort(var2);
         super.m.writer().writeByte(var3);
      } catch (IOException var5) {
      }

      this.sendMessage();
   }

   public final void doGivingDeferrent(int var1, int var2) {
      this.createMessage((byte)59);

      try {
         super.m.writer().writeInt(var1);
         super.m.writer().writeShort(var2);
      } catch (IOException var4) {
      }

      this.sendMessage();
   }

   public final void doRequestYourInfo(int var1) {
      System.out.println("doRequestYourInfo: " + var1);
      this.createMessage((byte)-22);
      this.writeInt(var1);
      this.sendMessage();
   }

   public final void doRequestBoardList(byte var1) {
      this.createMessage((byte)60);
      this.writeByte(var1);
      this.sendMessage();
   }

   public final void doBuyItem(short var1) {
      System.out.println("DEBUG BUYITEM: ParkService.doBuyItem(id=" + var1 + ") packet -38");
      this.createMessage((byte)-38);
      this.writeShort(var1);
      this.sendMessage();
   }

   public final void doFinishFishing(boolean var1, byte[] var2) {
      this.createMessage((byte)84);

      try {
         super.m.writer().writeBoolean(true);
         super.m.writer().writeByte(var2.length);

         for(int var4 = 0; var4 < var2.length; ++var4) {
            super.m.writer().writeByte(var2[var4]);
         }
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void doCauCaXong() {
      this.createMessage((byte)85);
      this.sendMessage();
   }

   public final void doInviteToMyHome(int var1, int var2) {
      this.createMessage((byte)-68);

      try {
         super.m.writer().writeByte(var1);
         super.m.writer().writeInt(var2);
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void doCustomPopup(int var1, int var2, int var3) {
      this.createMessage((byte)-77);

      try {
         super.m.writer().writeInt(var1);
         super.m.writer().writeByte(var2);
         super.m.writer().writeByte(var3);
      } catch (Exception var5) {
      }

      this.sendMessage();
   }

   public final void doBossShop(int var1, int var2, int var3) {
      System.out.println("DEBUG BOSS_SHOP(-78): idBoss=" + var1 + " shopByte=" + var2 + " optionIdx=" + var3);
      this.createMessage((byte)-78);

      try {
         super.m.writer().writeInt(var1);
         super.m.writer().writeByte(var2);
         super.m.writer().writeShort(var3);
      } catch (Exception var5) {
      }

      this.sendMessage();
   }

   public final void doGetDropPart(int var1) {
      this.createMessage((byte)89);

      try {
         super.m.writer().writeByte(0);
         super.m.writer().writeInt(var1);
      } catch (Exception var3) {
      }

      this.sendMessage();
   }

   public final void doRequestWedding(int var1, int var2) {
      System.out.println("doRequestWedding");
      this.createMessage((byte)93);

      try {
         super.m.writer().writeByte(var1);
         super.m.writer().writeByte(var2);
      } catch (Exception var4) {
         var4.printStackTrace();
      }

      this.sendMessage();
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/ParkService.cs
using System;
using System.IO;

public class ParkService
{
	protected static ParkService instance;

	public static ParkService gI()
	{
		if (instance == null)
		{
			instance = new ParkService();
		}
		return instance;
	}

	public void send(Message m)
	{
		Session_ME.gI().sendMessage(m);
		m.cleanup();
	}

	public void doJoinPark(int roomID, int boardID)
	{
		Canvas.startWaitDlg();
		Message message = new Message((sbyte)50);
		try
		{
			message.writer().writeByte((sbyte)roomID);
			message.writer().writeByte((sbyte)boardID);
			message.writer().writeShort((short)LoadMap.xJoinCasino);
			message.writer().writeShort((short)LoadMap.yJoinCasino);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doMove(int x, int y, int direct)
	{
		Message message = new Message((sbyte)54);
		try
		{
			message.writer().writeShort((short)x);
			message.writer().writeShort((short)y);
			message.writer().writeByte((sbyte)direct);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void chatToBoard(string text)
	{
		Message message = null;
		message = ((GameMidlet.CLIENT_TYPE != 10) ? new Message((sbyte)55) : new Message((sbyte)77));
		try
		{
			message.writer().writeUTF(text);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doRequestAddFriend(int idUser)
	{
		Message message = new Message((sbyte)(-21));
		try
		{
			message.writer().writeInt(idUser);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doAddFriend(int iddb, bool b)
	{
		Message message = new Message((sbyte)(-19));
		try
		{
			message.writer().writeInt(iddb);
			message.writer().writeBoolean(b);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doGiftGiving(int iddb, int i, int typeBuy)
	{
		Message message = new Message((sbyte)58);
		try
		{
			message.writer().writeInt(iddb);
			message.writer().writeShort((short)i);
			message.writer().writeByte((sbyte)typeBuy);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doGivingDeferrent(int idTo, int id)
	{
		Out.println("doGivingDeferrent: " + id);
		Message message = new Message((sbyte)59);
		try
		{
			message.writer().writeInt(idTo);
			message.writer().writeShort((short)id);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doRequestYourInfo(int iddb)
	{
		Message message = new Message((sbyte)(-22));
		try
		{
			message.writer().writeInt(iddb);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doRequestBoardList(sbyte roomId)
	{
		Message message = new Message((sbyte)60);
		try
		{
			message.writer().writeByte(roomId);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doBuyItem(short id)
	{
		Message message = new Message((sbyte)(-38));
		try
		{
			message.writer().writeShort(id);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doKick(sbyte roomId, sbyte boardId, int iddb)
	{
		Message message = new Message((sbyte)78);
		try
		{
			message.writer().writeByte(roomId);
			message.writer().writeByte(boardId);
			message.writer().writeInt(iddb);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doQuanCau()
	{
		Message m = new Message((sbyte)82);
		send(m);
	}

	public void doFinishFishing(bool isF, sbyte[] index)
	{
		Message message = new Message((sbyte)84);
		try
		{
			message.writer().writeBoolean(isF);
			message.writer().writeByte((sbyte)index.Length);
			for (int i = 0; i < index.Length; i++)
			{
				Canvas.test1 = Canvas.test1 + index[i] + ", ";
				message.writer().writeByte(index[i]);
			}
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doCauCaXong()
	{
		Message m = new Message((sbyte)85);
		send(m);
	}

	public void doStartFishing()
	{
		Message m = new Message((sbyte)86);
		send(m);
	}

	public void doInviteToMyHome(int type, int iddb)
	{
		Message message = new Message((sbyte)(-68));
		try
		{
			message.writer().writeByte((sbyte)type);
			message.writer().writeInt(iddb);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doCustomPopup(int idBoss, int idPopup, int ii)
	{
		Message message = new Message((sbyte)(-77));
		try
		{
			message.writer().writeInt(idBoss);
			message.writer().writeByte((sbyte)idPopup);
			message.writer().writeByte((sbyte)ii);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doBossShop(int idBoss1, int idShop, int ii)
	{
		Message message = new Message((sbyte)(-78));
		try
		{
			message.writer().writeInt(idBoss1);
			message.writer().writeByte((sbyte)idShop);
			message.writer().writeShort((short)ii);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doGetDropPart(int idDrop)
	{
		Message message = new Message((sbyte)89);
		try
		{
			message.writer().writeByte(0);
			message.writer().writeInt(idDrop);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doRequestWedding(int roomID, int boardID)
	{
		Message message = new Message((sbyte)93);
		try
		{
			message.writer().writeByte(roomID);
			message.writer().writeByte(boardID);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}
}
